optimum/bettertransformer/transformation.py

```diff
@@ -0,0 +1,190 @@
+    - Step 3: If yes, replace the `...Layer` module with the `...LayerBetterTransformer` modules
+    - Step 4: If not, yield an error.
+    - Step 5: Post process the potentially converted model by setting the `is_last_layer` attribute to `True` for the last `BetterTransformer` layer.
+    (done in `set_last_layer` function)
+
+    Args:
+        `model` (`torch.nn.Module`, **required**):
+            The input model to convert
+        `config` (`transformers.PreTrainedConfig`, **required**):
+            The configuration dictionary of the model
+    Returns:
+        The converted model
+    """
+    for name, module in model.named_children():
+        if len(list(module.children())) > 0:
+            replace_to_fast(module, config)
+
+        if hasattr(module, "is_decoder"):
+            # Decoders are not supported yet on Better Transformers
+            if module.is_decoder:
+                continue
+
+        class_name = module.__class__.__name__
+        maybe_fast_module = is_module_fast(class_name)
+
+        if not isinstance(maybe_fast_module, bool):
+            fast_module = maybe_fast_module(module, config)
+            model._modules[name] = fast_module
+        elif hasattr(module, "_hf_hook"):
+            # If the module has `accelerate` hooks, manually
+            # set the parameters on the `CPU` if there is
+            # any offload involved.
+            module = init_accelerate_hook(module)
+            model._modules[name] = module
+    return model
+
+
+def set_last_layer(model):
+    r"""
+    Iterates over the module list containing the `LayerBetterTransformer` modules. Sets the last layer's `is_last_layer`
+    attribute to `True`
+
+    Args:
+        `model` (`torch.nn.Module`, **required**):
+            The input converted model
+    Returns:
+        Returns `True` if it has succesfully set the attribute to `True`, otherwise return `False`.
+    """
+    dict_named_module = dict(model.named_modules())
+    sort_fn = lambda list_modules: [module.__class__.__name__ for module in list_modules]  # noqa: E731
+
+    for key in dict_named_module.keys():
+        if isinstance(dict_named_module[key], torch.nn.ModuleList) and all(
+            "LayerBetterTransformer" in module_name for module_name in sort_fn(dict_named_module[key])
+        ):
+            setattr(dict_named_module[key][-1], "is_last_layer", True)
+            return True
+    return False
+
+
+@check_if_pytorch_greater_112()
+class BetterTransformer(object):
+    r"""
+    A conversion wrapper that takes as an input the `transformers` model to be converted
+    and returns the converted `BetterTransformer` model. The `BetterTransformer` model is based on the `BetterTransformer`
+    recently released by PyTorch from its 1.12 version:
+    https://pytorch.org/blog/a-better-transformer-for-fast-transformer-encoder-inference/
+
+    Original PR from: https://github.com/huggingface/transformers/pull/19553 adapted and wrapped in
```

Comment:
Maybe we can put this out of the docstring in a single `#` comment so that it does not show in the doc

Reply:
Fixed in 1c1c6078588fc32f77370fb9b6d24be8bdf45117